diff --git a/dist/index.cjs b/dist/index.cjs
index ad67f5561ae09a571a66b7ea8aeb6751e8dc2083..3d3bf904b185a1b342a47a6b15d663e5faee88cf 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -759,7 +759,7 @@ class UnionTransformer {
 }
 
 class OpenApiTransformer {
-    constructor(versionSpecifics) {
+    constructor(versionSpecifics, openApiGenerator) {
         this.versionSpecifics = versionSpecifics;
         this.objectTransformer = new ObjectTransformer();
         this.stringTransformer = new StringTransformer();
@@ -774,6 +774,8 @@ class OpenApiTransformer {
         this.intersectionTransformer = new IntersectionTransformer();
         this.recordTransformer = new RecordTransformer();
         this.tupleTransformer = new TupleTransformer(versionSpecifics);
+        this.openApiGenerator = openApiGenerator;
+        this.lazyCount = 0;
     }
     transform(zodSchema, isNullable, mapItem, generateSchemaRef, defaultValue) {
         if (isZodType(zodSchema, 'ZodNull')) {
@@ -836,6 +838,19 @@ class OpenApiTransformer {
         if (isZodType(zodSchema, 'ZodDate')) {
             return this.versionSpecifics.mapNullableType('string', isNullable);
         }
+        if (isZodType(zodSchema, 'ZodLazy')) {
+            const schema = zodSchema.schema;
+
+            if (!('_lazyRefId' in zodSchema)) {
+                this.lazyCount++;
+                zodSchema._lazyRefId = `object_${this.lazyCount}`;
+                this.openApiGenerator.schemaRefs[zodSchema._lazyRefId] = this.openApiGenerator.generateSimpleSchema(
+                    schema.openapi(zodSchema._lazyRefId)
+                );
+            }
+
+            return this.openApiGenerator.schemaRefs[zodSchema._lazyRefId];
+        }
         const refId = Metadata.getRefId(zodSchema);
         throw new UnknownZodTypeError({
             currentSchema: zodSchema._def,
@@ -852,7 +867,7 @@ class OpenAPIGenerator {
         this.paramRefs = {};
         this.pathRefs = {};
         this.rawComponents = [];
-        this.openApiTransformer = new OpenApiTransformer(versionSpecifics);
+        this.openApiTransformer = new OpenApiTransformer(versionSpecifics, this);
         this.sortDefinitions();
     }
     generateDocumentData() {
