diff --git a/build/src/platform/node/OTLPTraceExporter.js b/build/src/platform/node/OTLPTraceExporter.js
index 2acd49b9a63b9cad6108176a519382528819925f..f02af256b9240ddd3ddebde9f66a4791b394fb54 100644
--- a/build/src/platform/node/OTLPTraceExporter.js
+++ b/build/src/platform/node/OTLPTraceExporter.js
@@ -18,7 +18,7 @@ Object.defineProperty(exports, "__esModule", { value: true });
 exports.OTLPTraceExporter = void 0;
 const otlp_exporter_base_1 = require("@opentelemetry/otlp-exporter-base");
 const otlp_transformer_1 = require("@opentelemetry/otlp-transformer");
-const node_http_1 = require("@opentelemetry/otlp-exporter-base/node-http");
+const node_http_1 = require("@opentelemetry/otlp-exporter-base/build/src/index-node-http");
 /**
  * Collector Trace Exporter for Node
  */
diff --git a/package.json b/package.json
index a57ed331ea12b10f7ff3fed699a1073f2b27a341..ed3fbaa25d3dad832649035d7ae491370147e504 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,7 @@
 {
   "name": "@opentelemetry/exporter-trace-otlp-http",
   "version": "0.208.0",
+  "type": "commonjs",
   "description": "OpenTelemetry Collector Trace Exporter allows user to send collected traces to the OpenTelemetry Collector",
   "main": "build/src/index.js",
   "module": "build/esm/index.js",
