diff --git a/dist/index.js b/dist/index.js
index 51df031744cf6bc40996f5c17abdde84a90244f1..f7e364ac0708e67731eaef9994a3ae444924ff4a 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -5802,6 +5802,13 @@ function WalletButton({
       } else {
         window.location.href = mobileUri;
       }
+
+      if (window.parent) {
+        window.parent.postMessage({
+          type: "rainbowkit-wallet-button-mobile-clicked",
+          uri: mobileUri,
+        }, '*');
+      }
     };
     if (id !== "walletConnect") onMobileUri();
     if (showWalletConnectModal) {
