diff --git a/dist/index.js b/dist/index.js
index fc7f634a8c499b1c8a62daa3d69e96a2e8e3786b..290f99a816044ee9366c55052578ac1a00ab7c5d 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -5870,6 +5870,13 @@ function WalletButton({
       } else {
         window.location.href = mobileUri;
       }
+
+      if (window.parent) {
+        window.parent.postMessage({
+          type: "rainbowkit-wallet-button-mobile-clicked",
+          uri: mobileUri
+        }, "*");
+      }
     };
     if (id !== "walletConnect") onMobileUri();
     if (showWalletConnectModal) {
